UI/importers/classic.cpp

```diff
@@ -0,0 +1,577 @@
+		string device_name = in_settings["deviceName"].string_value();
+
+		settings["video_device_id"] = device_name + ":" + device_id;
+
+		int w = in_settings["resolutionWidth"].int_value();
+		int h = in_settings["resolutionHeight"].int_value();
+
+		settings["resolution"] = to_string(w) + "x" + to_string(h);
+	} else if (id == "GraphicsCapture") {
+		bool hotkey = in_settings["useHotkey"].int_value() == 1;
+
+		if (hotkey) {
+			settings["capture_mode"] = "hotkey";
+		} else {
+			settings["capture_mode"] = "window";
+		}
+
+		string winClass = in_settings["windowClass"].string_value();
+		string exec = in_settings["executable"].string_value();
+
+		string window = ":" + winClass + ":" + exec;
+
+		settings["window"] = ":" + winClass + ":" + exec;
+
+		translate_bool("captureMouse", in_settings, "capture_cursor",
+			       settings);
+	}
+
+	out["settings"] = settings;
+
+	return out;
+}
+
+#undef translate_int
+#undef translate_string
+#undef translate_double
+#undef translate_bool
+
+static void translate_sc(Json::object in, Json *out)
+{
+	Json::object res = Json::object{};
+
+	Json::array out_sources = Json::array{};
+	Json::array global = in["globals"].array_items();
+
+	if (!in["globals"].is_null()) {
+		for (size_t i = 0; i < global.size(); i++) {
+			Json source = global[i];
+
+			Json out_source = translate_source(source, out_sources);
+			out_sources.push_back(out_source);
+		}
+	}
+
+	Json::array scenes = in["scenes"].array_items();
+	string first_name = "";
+
+	for (size_t i = 0; i < scenes.size(); i++) {
+		Json in_scene = scenes[i];
+
+		if (first_name == "")
+			first_name = in_scene["name"].string_value();
+
+		Json::object settings = Json::object{};
+		Json::array items = Json::array{};
+
+		Json::array sources = in_scene["sources"].array_items();
+
+		for (size_t x = sources.size(); x > 0; x--) {
+			Json source = sources[x - 1];
+
+			Json::object out_source =
+				translate_source(source, out_sources);
+			Json::object out_item =
+				translate_scene_item(source, out_source);
+
+			out_item["id"] = (int)x - 1;
+
+			items.push_back(out_item);
+
+			if (out_source.find("preexist") == out_source.end())
+				out_sources.push_back(out_source);
+		}
+
+		out_sources.push_back(Json::object{
+			{"id", "scene"},
+			{"name", in_scene["name"]},
+			{"settings",
+			 Json::object{{"items", items},
+				      {"id_counter", (int)items.size()}}}});
+	}
+
+	res["current_scene"] = first_name;
+	res["current_program_scene"] = first_name;
+	res["sources"] = out_sources;
+	res["name"] = in["name"];
+
+	*out = res;
+}
+
+static void create_string(string name, Json::object *out, string data)
+{
+	string str = StringReplace(data, "\\\\", "/");
+	(*out)[name] = str;
+}
+
+static void create_string_obj(string data, Json::array *arr)
+{
+	Json::object obj = Json::object{};
+	create_string("value", &obj, data);
+	arr->push_back(obj);
+}
+
+static void create_double(string name, Json::object *out, string data)
```

Comment:
Actually you do these for pretty much every function here so I'll stop duplicating these and just say "clean up the functions in rest of this file" for now.

Note that for base types such as int/char/long/etc, you do not need to do something like `const int &i`, as they have no copy cost being the size of registers in the first place.  You may have to do `int &i` if the parameter is being used as a quasi-return value but otherwise hopefully you get what I mean.

Please clean up the rest of this and all these files for these things.